trustymail/trustymail.py

```diff
@@ -328,12 +318,104 @@ def dmarc_scan(resolver, domain):
+                    if '0' in values and '1' in values:
+                        msg = 'fo tag values 0 and 1 are mutually exclusive'
+                        handle_syntax_error('[DMARC]', domain, '{0}'.format(msg))
+                    for value in values:
+                        if value not in ['0', '1', 'd', 's']:
+                            msg = 'Unknown DMARC fo tag value {0}'.format(value)
+                            handle_syntax_error('[DMARC]', domain, '{0}'.format(msg))
+                            domain.valid_dmarc = False
+                elif tag == 'rf':
+                    values = tag_dict[tag].split(':')
+                    for value in values:
+                        if value not in ['afrf']:
+                            msg = 'Unknown DMARC rf tag value {0}'.format(value)
+                            handle_syntax_error('[DMARC]', domain, '{0}'.format(msg))
+                            domain.valid_dmarc = False
+                elif tag == 'ri':
+                    try:
+                        int(tag_dict[tag])
+                    except ValueError:
+                        msg = 'Invalid DMARC ri tag value: {0} - must be an integer'.format(tag_dict[tag])
+                        handle_syntax_error('[DMARC]', domain, '{0}'.format(msg))
+                        domain.valid_dmarc = False
+                elif tag == 'pct':
+                    try:
+                        pct = int(tag_dict[tag])
+                        if pct < 0 or pct > 100:
+                            msg = 'Error: invalid DMARC pct tag value: {0} - must be an integer between ' \
+                                  '0 and 100'.format(tag_dict[tag])
+                            handle_syntax_error('[DMARC]', domain, '{0}'.format(msg))
+                            domain.valid_dmarc = False
+                        domain.dmarc_pct = pct
+                    except ValueError:
+                        msg = 'invalid DMARC pct tag value: {0} - must be an integer'.format(tag_dict[tag])
+                        handle_syntax_error('[DMARC]', domain, '{0}'.format(msg))
+                        domain.valid_dmarc = False
+                elif tag == 'rua' or tag == 'ruf':
+                    uris = tag_dict[tag].split(',')
+                    for uri in uris:
+                        # mailto: is currently the only type of DMARC URI
+                        mailto_matches = MAILTO_REGEX.findall(uri)
+                        if len(mailto_matches) != 1:
+                            msg = 'Error: {0} is an invalid DMARC URI'.format(uri)
+                            handle_syntax_error('[DMARC]', domain, '{0}'.format(msg))
+                            domain.valid_dmarc = False
+                        else:
+                            email_address = mailto_matches[0]
+                            email_domain = email_address.split('@')[-1]
+                            if email_domain.lower() != domain.domain_name.lower():
```

Comment:
@seanthegeek, what if `email_domain` is a subdomain of `domain_name`?  Does this case need to be treated separately?  I don't think so, but I think you're read the standard more closely than I have.

`dhs.gov` sends email to an `hq.dhs.gov` address, but it lacks the special DNS record and so it fails the check.

Reply:
No separate check is necessary. When I originally wrote this code for my own DMARC heck script, DHS.gov was one of my test cases. When this message fired, I looked for who I could report it to, which led me to this project :)

Reply:
Got it.